clustered/integration-test/src/test/java/org/ehcache/clustered/LeaseTest.java

```diff
@@ -0,0 +1,199 @@
+public class LeaseTest extends ClusteredTests {
+  private static final String STRIPE_SEPARATOR = ",";
+
+  private static final String RESOURCE_CONFIG =
+          "<config xmlns:ohr='http://www.terracotta.org/config/offheap-resource'>"
+                  + "<ohr:offheap-resources>"
+                  + "<ohr:resource name=\"primary-server-resource\" unit=\"MB\">64</ohr:resource>"
+                  + "</ohr:offheap-resources>"
+                  + "</config>\n"
+                  + "<service xmlns:lease='http://www.terracotta.org/service/lease'>"
+                  + "<lease:connection-leasing>"
+                  + "<lease:lease-length unit='seconds'>5</lease:lease-length>"
+                  + "</lease:connection-leasing>"
+                  + "</service>";
+
+  @ClassRule
+  public static Cluster CLUSTER =
+          newCluster().in(new File("build/cluster")).withServiceFragment(RESOURCE_CONFIG).build();
+
+  private final List<TCPProxy> proxies = new ArrayList<>();
+
+  @BeforeClass
+  public static void waitForActive() throws Exception {
+    CLUSTER.getClusterControl().waitForActive();
+  }
+
+  @After
+  public void after() {
+    proxies.forEach(TCPProxy::stop);
+  }
+
+  @Parameterized.Parameters
+  public static ResourcePoolsBuilder[] data() {
+    return new ResourcePoolsBuilder[]{
+            ResourcePoolsBuilder.newResourcePoolsBuilder()
+                    .with(clusteredDedicated("primary-server-resource", 1, MemoryUnit.MB)),
+            ResourcePoolsBuilder.newResourcePoolsBuilder()
+                    .heap(10, EntryUnit.ENTRIES)
+                    .with(clusteredDedicated("primary-server-resource", 1, MemoryUnit.MB))
+    };
+  }
+
+  @Parameterized.Parameter
+  public ResourcePoolsBuilder resourcePoolsBuilder;
+
+  @Test
+  public void leaseExpiry() throws Exception {
+    URI connectionURI = getProxyURI();
+
+    CacheManagerBuilder<PersistentCacheManager> clusteredCacheManagerBuilder
+            = CacheManagerBuilder.newCacheManagerBuilder()
+            .with(ClusteringServiceConfigurationBuilder.cluster(connectionURI.resolve("/crud-cm"))
+                    .timeouts(TimeoutsBuilder.timeouts()
+                            .connection(Duration.ofSeconds(20)))
+                    .autoCreate()
+                    .defaultServerResource("primary-server-resource"));
+    PersistentCacheManager cacheManager = clusteredCacheManagerBuilder.build(false);
+    cacheManager.init();
+
+    CacheConfiguration<Long, String> config = CacheConfigurationBuilder.newCacheConfigurationBuilder(Long.class, String.class,
+            resourcePoolsBuilder).build();
+
+    Cache<Long, String> cache = cacheManager.createCache("clustered-cache", config);
+    cache.put(1L, "The one");
+    cache.put(2L, "The two");
+    cache.put(3L, "The three");
+    assertThat(cache.get(1L), equalTo("The one"));
+    assertThat(cache.get(2L), equalTo("The two"));
+    assertThat(cache.get(3L), equalTo("The three"));
+
+    setDelay(6000);
```

Comment:
6 seconds is too long to wait in a test. Can't be make the lease expiring faster? (or everything to rely on a time source)